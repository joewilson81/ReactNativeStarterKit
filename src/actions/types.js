export const BUTTON_PUSH = 'button_push';
