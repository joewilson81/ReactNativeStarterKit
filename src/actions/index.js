import { BUTTON_PUSH } from './types';

export const buttonPush = () => {
  return { type: BUTTON_PUSH };
}
