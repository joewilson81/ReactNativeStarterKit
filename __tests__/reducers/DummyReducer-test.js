import reducer from '../../src/reducers/DummyReducer';
import * as types from '../../src/actions/types';

describe('DummyReducer', () => {
  const INITIAL_STATE = { buttonPressed: false };

  it('should return the initial state', () => {
    // When the app first mounts, the state is undefined
    // Simulate this with an empty action object and we should get
    // a response that looks exactly like our INITIAL_STATE object.
    const previousState = undefined;
    const action = {};

    expect(reducer(previousState, action)).toMatchSnapshot();
  });

  it('Should handle BUTTON_PUSH', () => {
    const previousState = INITIAL_STATE;
    const action = { type: types.BUTTON_PUSH };

    expect(reducer(previousState, action)).toMatchSnapshot();
  });
});
