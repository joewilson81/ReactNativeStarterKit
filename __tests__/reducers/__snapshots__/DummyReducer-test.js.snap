// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyReducer Should handle BUTTON_PUSH 1`] = `
Object {
  "buttonPressed": true,
}
`;

exports[`DummyReducer should return the initial state 1`] = `
Object {
  "buttonPressed": false,
}
`;
